lightning/src/ln/channel.rs

```diff
@@ -2302,7 +2302,21 @@ impl<SP: Deref> InitialRemoteCommitmentReceiver<SP> for FundedChannel<SP> where
 	}
 }
 
-impl<SP: Deref> PendingV2Channel<SP> where SP::Target: SignerProvider {
+/// A channel struct implementing this trait can perform V2 transaction negotiation,
+/// either at channel open or during splicing.
+/// Accessors return a Result, because [`FundedChannel`] can act like a TX constructor,
+/// but not always (only during splicing).
```

Comment:
Hmm.... we could avoid needing `Result` by creating a temporary `RenegotiatingFunding` wrapper around `&FundedChannel` that implements `FundingTxConstructorV2`. It could be obtained using a `as_renegotiating_funding` method on `FundedChannel`. Basically, push the `Result` / `Option` check to when calling `as_renegotiating_funding` instead of all the other methods.

Reply:
Done with a wrapper as suggested.